README.md

```diff
@@ -1 +1,50 @@
-# bounded_counter
\ No newline at end of file
+Bounded Counter – Count-min Sketch frequency estimation for large sets
+======================================================================
+
+Library implementing [Count-min Sketch](https://en.wikipedia.org/wiki/Count%E2%80%93min_sketch)
+data structure used to estimate frequencies of elements in massive data sets with fixed memory footprint.
+
+Contains implementation of the basic and conservative-update Count-min Sketch table, as well as our own implementation
+using logarithmic probabilistic counter for reducing the space requirement (*logcons1024*).
+
+A [hyperloglog](https://github.com/ascv/HyperLogLog) counter is used to estimate the cardinality of the set of elements (how many distinct elements are tracked).
+
+Parameters
+----------
+
+-   **Width**: The number of columns (hash buckets) in a single row of the table. Must be a power of 2.
+    Significantly affects precision and memory footprint. For precise results, this should be no smaller than one
+    order of magnitude away from the cardinality of the set.
+    For significantly smaller widths, deterioration will occur.
+    For instance, to store all bigrams in English Wikipedia (1 857 420 106 bigrams, 179 413 989 unique),
+    good results can be achieved with a width of 67 108 864 (2^26) (37%).
+-   **Depth**: Number of rows, significant for the reliability of the result. Linearly affects speed of the
```

Comment:
Way too complicated. The API should accept a memory limit, in (mega/giga)bytes.

We can keep these width/depth params exposed, for power users, but not by default.